app/models/agents/naive_bayes_agent.rb

```diff
@@ -0,0 +1,509 @@
+require 'yaml'
+
+# NBayes from Oasic Technologies
+# https://github.com/oasic/nbayes
+#
+# Robust implementation of NaiveBayes:
+# - using log probabilities to avoid floating point issues
+# - Laplacian smoothing for unseen tokens
+# - allows binarized or standard NB
+# - allows Prior distribution on category to be assumed uniform (optional)
+# - generic to work with all types of tokens, not just text
+
+
+module NBayes
```

Comment:
I had to change 1 line in the NBayes gem to make the saving to Agent memory work easily. So it might be possible to work around this. There is a comment in the code towards the bottom of the NBayes module if you want to take a look (line 345). 

As I stated above, I'm new to Ruby, so I'm a bit unsure about how all this works. Same goes for rspec. But I will give it a shot.